spec.md

```diff
@@ -363,6 +368,7 @@ on the definition of OPTIONAL.
   for more information.
 - Constraints:
   - OPTIONAL
+  - If present, MUST be a non-empty URI-reference
```

Comment:
I'm more skeptical of the value of a URI-reference here; I sort of understand the thinking that we want to allow a subset of general strings as `source` attributes even if they don't conform to the
```URI           = scheme ":" hier-part [ "?" query ] [ "#" fragment ]```
production. Since `schemaurl` is OPTIONAL rather than required, why not require that the schema URL be an actual URI and not relative to some (unknown) base?

Reply:
@evankanderson I think that's part of #475.